vulture_os/portal/system/authentications.py

```diff
@@ -155,73 +161,68 @@ def authenticate_on_backend(self, backend):
                                                                           self.credentials[0], self.credentials[1],
-                                                                          oauth2_token,
+                                                                          self.oauth2_token,
                                                                           authentication_results,
                                                                           self.workflow.authentication.auth_timeout)
         logger.debug("AUTH::register_user: Authentication results successfully written in Redis portal session")
-
-        return portal_cookie, oauth2_token
+        return portal_cookie, self.oauth2_token
 
     def register_sso(self, backend_id):
         username = self.redis_portal_session.keys['login_' + backend_id]
-        oauth2_token = self.redis_portal_session.keys['oauth2_' + backend_id]
-        self.redis_oauth2_session = REDISOauth2Session(self.redis_portal_session.handler, "oauth2_" + oauth2_token)
+        self.oauth2_token = self.redis_portal_session.keys['oauth2_' + backend_id]
+        self.redis_oauth2_session = REDISOauth2Session(self.redis_portal_session.handler, "oauth2_" + self.oauth2_token)
         logger.debug("AUTH::register_sso: Redis oauth2 session successfully retrieven")
         password = self.redis_portal_session.getAutologonPassword(self.workflow.id, backend_id, username)
         logger.debug("AUTH::register_sso: Password successfully retrieven from Redis portal session")
         portal_cookie = self.redis_portal_session.register_sso(self.workflow.authentication.auth_timeout,
                                                                backend_id, str(self.workflow.id),
                                                                str(self.workflow.get_redirect_uri()), username,
-                                                               oauth2_token)
+                                                               self.oauth2_token)
         logger.debug("AUTH::register_sso: SSO informations successfully written in Redis for user {}".format(username))
         self.credentials = [username, password]
         if self.double_authentication_required():
             self.redis_portal_session.register_doubleauthentication(self.workflow.id,
                                                                     self.workflow.authentication.otp_repository)
             logger.debug("AUTH::register_sso: DoubleAuthentication required : "
                          "successfully written in Redis for user '{}'".format(username))
-        return portal_cookie, oauth2_token
+        return portal_cookie, self.oauth2_token
+
+    def register_openid(self, openid_token, **kwargs):
+        # Generate a new OAuth2 token
+        #self.oauth2_token = Uuid4().generate()
+        # Register it into session
```

Comment:
```suggestion
        # Register it into session using existing oauth2 token
```

Reply:
I prefer to keep this line, as a reminder for later improvement - we should regenerate a new token each time